src/cdn_definitions/_impl/__init__.py

```diff
@@ -1,26 +1,103 @@
+    return yaml.load(
+        data.text if isinstance(data, requests.Response) else data, yaml.SafeLoader
+    )
 
-def load_data():
-    # Load data from first existing of these
-    candidate_paths = [
-        os.path.join(os.path.dirname(os.path.dirname(__file__)), "data.yaml"),
-        "/usr/share/cdn-definitions/data.yaml",
-    ]
 
-    # If env var is set, it takes highest precedence
-    if "CDN_DEFINITIONS_PATH" in os.environ:
-        candidate_paths.insert(0, os.environ["CDN_DEFINITIONS_PATH"])
+def get_remote_data(url):
+    """
+    Creates a requests session with retries. If the request was successful, returns the response.
+    """
+    retry_strategy = Retry(
+        status_forcelist=[429, 500, 502, 503, 504],
+    )
+    adapter = requests.adapters.HTTPAdapter(max_retries=retry_strategy)
+    session = requests.Session()
+    session.mount(url, adapter)
 
-    existing_paths = [p for p in candidate_paths if os.path.exists(p)]
+    response = session.get(url, timeout=10)
 
-    _, ext = os.path.splitext(existing_paths[0])
-    if ext.lower() == ".json":
-        return json.load(open(existing_paths[0]))
+    if response.raise_for_status() is None:
+        return response
+
+
+def get_local_data(data_path):
+    """
+    Given a valid path to a file, returns the contents of the file.
+    """
+    with open(data_path, "rb") as data_file:
+        return data_file.read()
+
+
+def get_data(data_paths):
+    """
+    Returns a dict of the data at a URL or local file path.
+    If data cannot be loaded, raises a RuntimeError.
+    """
+    for path in data_paths:
+        _, ext = os.path.splitext(path)
 
-    return yaml.load(open(existing_paths[0]), yaml.SafeLoader)
+        if path.startswith("http"):
+            return parsed(get_remote_data(path), ext)
+
+        if os.path.exists(path):
+            return parsed(get_local_data(path), ext)
+    raise RuntimeError("Could not load data")
+
+
+def load_data(source=None):
+    """Loads data from YAML or JSON format into a dictionary.
+
+    Args:
+        source (str, optional): A local path or URL to a JSON or YAML data file.
+
+    Returns:
+        dict: The data from the local path or URL in the form of a dictionary.
+
+    Raises:
+        RuntimeError: If all attempted data sources are invalid, a RuntimeError will be raised.
+
+    """
+    if source is None:
+        # Load data from first existing of these
+        candidate_paths = [
+            os.path.join(os.path.dirname(os.path.dirname(__file__)), "data.yaml"),
+            "/usr/share/cdn-definitions/data.yaml",
+        ]
+
+        # If env var is set, it takes highest precedence
+        if "CDN_DEFINITIONS_PATH" in os.environ:
+            candidate_paths.insert(0, os.environ["CDN_DEFINITIONS_PATH"])
+
+        return get_data(candidate_paths)
+    return get_data([source])
+
+
+def load_schema():
+    """
+    Loads the `schema.json` file provided with the cdn-definitions package into a dictonary.
```

Comment:
```suggestion
    Loads the `schema.json` file provided with the cdn-definitions package into a dictionary.
```

Reply:
This comment made me realize that I forgot to rephrase this line that suggests that this method will always return a dict, as you pointed out, here: 

>  Not a big deal, but the function name and doc string aren't actually true since JSON or YAML files can contain kinds of data other than a dict. e.g. a file containing literally 1 is valid JSON and valid YAML and would cause this function to return an integer. I'd have named it something like parsed since it's parsing the input as JSON or YAML but not converting it to any particular type.


 I changed `load_data`'s and `load_schema`'s doc strings to more generally state that they return the data coerced into a python object instead of a dict.